internal/transformer/classes/generators/dockerfilegenerators/PHPdockerfilegenerator.go

```diff
@@ -0,0 +1,224 @@
+			logrus.Errorf("Error while converting the port from string to int : %s", err)
+			return port
+		}
+		return int32(detectedPort)
+	}
+	defer confFile.Close()
+	return port
+}
+
+// detectConfFiles detects if conf files are present or not
+func detectConfFiles(dir string) ([]string, error) {
+	var confFilesPaths []string
+	confFiles, err := common.GetFilesByExt(dir, []string{confExt})
+	if err != nil {
+		logrus.Debugf("Could not find conf files %s", err)
+		return confFilesPaths, err
+	}
+	for _, confFilePath := range confFiles {
+		confFileRelPath, err := filepath.Rel(dir, confFilePath)
+		if err != nil {
+			logrus.Errorf("Unable to resolve conf file path %s as rel path : %s", confFilePath, err)
+			continue
+		}
+		confFilesPaths = append(confFilesPaths, confFileRelPath)
+	}
+	return confFilesPaths, nil
+}
+
+// DirectoryDetect runs detect in each sub directory
+func (t *PHPDockerfileGenerator) DirectoryDetect(dir string) (namedServices map[string]transformertypes.ServicePlan, unnamedServices []transformertypes.TransformerPlan, err error) {
+	dirEntries, err := os.ReadDir(dir)
+	if err != nil {
+		logrus.Errorf("Error while trying to read directory : %s", err)
+		return nil, nil, err
+	}
+	for _, de := range dirEntries {
+		if filepath.Ext(de.Name()) != phpExt {
+			continue
+		}
+		unnamedServices = []transformertypes.TransformerPlan{{
+			Mode:              t.Config.Spec.Mode,
+			ArtifactTypes:     []transformertypes.ArtifactType{artifacts.ContainerBuildArtifactType},
+			BaseArtifactTypes: []transformertypes.ArtifactType{artifacts.ContainerBuildArtifactType},
+			Paths: map[string][]string{
+				artifacts.ProjectPathPathType: {dir},
+			},
+		}}
+		return nil, unnamedServices, nil
+	}
+	return nil, nil, nil
+}
+
+// Transform transforms the artifacts
+func (t *PHPDockerfileGenerator) Transform(newArtifacts []transformertypes.Artifact, oldArtifacts []transformertypes.Artifact) ([]transformertypes.PathMapping, []transformertypes.Artifact, error) {
+	pathMappings := []transformertypes.PathMapping{}
+	artifactsCreated := []transformertypes.Artifact{}
+	for _, a := range newArtifacts {
+		if a.Artifact != artifacts.ServiceArtifactType {
+			continue
+		}
+		relSrcPath, err := filepath.Rel(t.Env.GetEnvironmentSource(), a.Paths[artifacts.ProjectPathPathType][0])
+		if err != nil {
+			logrus.Errorf("Unable to convert source path %s to be relative : %s", a.Paths[artifacts.ProjectPathPathType][0], err)
+		}
+		var sConfig artifacts.ServiceConfig
+		err = a.GetConfig(artifacts.ServiceConfigType, &sConfig)
+		if err != nil {
+			logrus.Errorf("unable to load config for Transformer into %T : %s", sConfig, err)
+			continue
+		}
+		sImageName := artifacts.ImageName{}
+		err = a.GetConfig(artifacts.ImageNameConfigType, &sImageName)
+		if err != nil {
+			logrus.Debugf("unable to load config for Transformer into %T : %s", sImageName, err)
+		}
+		var detectedPorts []int32
+		var phpConfig PhpTemplateConfig
+		var confFilePath string
```

Comment:
Do we need the confFilePath? Why not directly assign to phpConfig.ConfFile?

Reply:
Yes, we don't need the confFilePath.